homeassistant/components/binary_sensor/insteon_plm.py

```diff
@@ -0,0 +1,96 @@
+"""
+Support for INSTEON dimmers via PowerLinc Modem.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/insteon_plm/
+"""
+import logging
+import asyncio
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.loader import get_component
+
+DEPENDENCIES = ['insteon_plm']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the INSTEON PLM device class for the hass platform."""
+    _LOGGER.info('Provisioning Insteon PLM Binary Sensors')
+
+    plm = hass.data['insteon_plm']
+
+    def async_plm_binarysensor_callback(device):
+        """New device detected from transport."""
+        name = device['address']
+        address = device['address_hex']
+
+        _LOGGER.info('New INSTEON PLM binary sensor device: %s (%s)',
+                     name, address)
+        hass.async_add_job(async_add_devices(
+            [InsteonPLMBinarySensorDevice(hass, plm, address, name)]))
+
+    criteria = {'capability': 'binary_sensor'}
+    plm.protocol.devices.add_device_callback(
+        async_plm_binarysensor_callback, criteria)
+
+    new_binarysensors = []
+    yield from async_add_devices(new_binarysensors)
+
+
+class InsteonPLMBinarySensorDevice(BinarySensorDevice):
+    """A Class for an Insteon device."""
+
+    def __init__(self, hass, plm, address, name):
+        """Initialize the binarysensor."""
+        self._hass = hass
+        self._plm = plm.protocol
+        self._address = address
+        self._name = name
+
+        self._plm.add_update_callback(
+            self.async_binarysensor_update, {'address': self._address})
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def address(self):
+        """Return the the address of the node."""
+        return self._address
+
+    @property
+    def name(self):
+        """Return the the name of the node."""
+        return self._name
+
+    @property
+    def sensor_class(self):
+        """Sensor class is unknown."""
+        return
+
+    @property
+    def is_on(self):
+        """Return the boolean response if the node is on."""
+        sensorstate = self._plm.get_device_attr(self._address, 'sensorstate')
```

Comment:
This method does not cause any IO, it's just probing a private data structure within the insteon_plm class ([ref](https://github.com/nugget/python-insteonplm/blob/master/insteonplm/protocol.py#L815))